packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,287 @@
+        saveInfo,
+    );
+
+    await finalState.testTreeProvider.ensureSynchronized();
+    validateTree(provider.trees[0], [initialTreeState]);
+
+    if (validateAnchor) {
+        const expectedPath: UpPath = {
+            parent: {
+                parent: undefined,
+                parentIndex: 0,
+                parentField: rootFieldKeySymbol,
+            },
+            parentField: brand("foo"),
+            parentIndex: 1,
+        };
+        const anchorPath = tree.locate(firstAnchor);
+        assert(compareUpPaths(expectedPath, anchorPath));
+    }
+    return finalState;
+}
+
+function applyFuzzChange(tree: ISharedTree, contents: FuzzChange, transactionResult:TransactionResult): void {
+    const index = contents.index;
+    const nodeField = contents.field;
+    switch (contents.fuzzType) {
+        case "insert":
+            if (index !== undefined && nodeField !== undefined) {
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(contents.parent, nodeField);
+                    field.insert(
+                        index,
+                        singleTextCursor({ type: brand("Test"), value: contents.value }),
+                    );
+                    return transactionResult;
+                });
+            }
+            break;
+        case "delete":
+            if (index !== undefined && nodeField !== undefined) {
+                const parent = contents.parent?.parent;
+                const delField = contents.parent?.parentField;
+                const parentIndex = contents.parent?.parentIndex;
+                if (delField !== undefined && parent !== undefined && parentIndex !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        const field = editor.sequenceField(parent, delField);
+                        field.delete(parentIndex, 1);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        case "setPayload":
+            if (index !== undefined && nodeField !== undefined) {
+                const path = contents.parent;
+                if (path !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        editor.setValue(path, contents.value);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+enum FuzzTestType {
+    Basic = 0,
+    Abort = 1,
+    AnchorStability = 2,
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    const random = makeRandom(0)
+    const testBatchSize = 3;
+    describeFuzz(title, ({ testCount }) => {
+        function runFuzzTest (
+            opGenerator: () => AsyncGenerator<Operation, FuzzTestState>,
+            fuzzTestType: FuzzTestType,
+            stepSize: number,
```

Comment:
Names of the form `fooSize` are generally mean to represent the size of `foo`, which is the way you're using it for `batchSize`. Here however we're not trying to represent the size of a step. I recommend renaming it to something like `opsPerRun`.